src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs

```diff
@@ -66,9 +66,17 @@ public override int Execute()
 
             if (!_appliedCommand.HasOption("no-restore"))
             {
-                // Create a Dependency Graph file for the project
-                tempDgFilePath = Path.GetTempFileName();
-                GetProjectDependencyGraph(projectFilePath, tempDgFilePath);
+                
+                try
+                {
+                    // Create a Dependency Graph file for the project
+                    tempDgFilePath = Path.GetTempFileName();
+                    GetProjectDependencyGraph(projectFilePath, tempDgFilePath);
```

Comment:
The call to `GetProjectDependencyGraph` should be outside of the try/catch block, so it doesn't accidentally catch exceptions from that method and think that it failed to create a temp path.